src/neo/IO/Json/JPathToken.cs

```diff
@@ -299,9 +303,10 @@ static IEnumerable<JObject> GetElements(JArray array, int[] indexes)
             if (maxDepth <= 0) throw new InvalidOperationException();
             --maxDepth;
             objects = objects.OfType<JArray>().SelectMany(p => GetElements(p, indexes)).ToArray();
+            if (objects.Length > maxObjects) throw new InvalidOperationException(nameof(maxObjects));
         }
 
-        private static void DescentRange(ref JObject[] objects, ref int maxDepth, int start, int end)
+        private static void DescentRange(ref JObject[] objects, int start, int end, ref int maxDepth, int maxObjects)
```

Comment:
Why do you change the order of parameters?

Reply:
Fixed